packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesModule.kt

```diff
@@ -194,13 +200,13 @@ class UpdatesModule : Module(), IUpdatesEventManagerObserver {
     }
 
     AsyncFunction("readLogEntriesAsync") { maxAge: Long, promise: Promise ->
```

Comment:
IIRC there was a way to use the built in Module coroutine scope, not sure if it'd be applicable here yet though (I assume it's probably the end state and requires `readLogEntries` to be a suspending func so it'll be a later refactor)

```suggestion
    AsyncFunction("readLogEntriesAsync") Coroutine { maxAge: Long, ->
```

Reply:
Probably better to just do that now instead of making a new PR. I'll update this

Reply:
One challenge I ran into was knowing if having two separate coroutine scopes (one from the module and one from the singleton controller) could have any downsides. But I think it was just my lack of depth of knowledge about coroutines that made it a challenge.

Reply:
That would be fine. Scopes are just a way to do structured concurrency. They give you a way to manage errors, cancellation, order, etc.

Reply:
I'm updating `readLogEntriesAsync` but I'll leave `clearLogEntriesAsync` for now. That passes a callback through 3 functions so ideally we should refactor those

Reply:
Ah yes, our good ol' callback hell.